### PR TITLE
Reduce oxygen and damage player in bubble columns

### LINKED ISSUE #11: Apply damage to bubble colums
Using bubble colums should damage the player like waterfall does, both upwards and downwards.

### PR DESCRIPTION
**Pull request by Boy0000**
Reduces the players oxygen whilst in a bubble column, and damages them when it runs out

plox dont shit on it too hard offz :)

solves #11

### CODE REVIEW

File: src/main/kotlin/org/cultofclang/bonehurtingjuice/Extensions.kt [hunk 1]
```diff
@@ -16,6 +18,11 @@ internal val Block.isFlowing: Boolean
         val blockData = blockData
         return (blockData is Levelled) && blockData.level >= 8
     }
+internal val Block.isBubbleColumn: Boolean
+    get() {
+        val blockData = blockData
```
Comment (0ffz) [inline src/main/kotlin/org/cultofclang/bonehurtingjuice/Extensions.kt]: why the separate statements here?

```kotlin
internal val Block.isBubbleColumn: Boolean = blockData is BubbleColumn
```
should be enough

File: src/main/kotlin/org/cultofclang/bonehurtingjuice/MoveListener.kt [hunk 2]
```diff
@@ -58,7 +62,7 @@ internal object MoveListener : Listener {
         player.location.findLocationAround(radius = 1, scale = 0.25) {
             val inBlock = it.block
             val higherBlock = it.add(0.0, 4.0, 0.0).block
-            inBlock.isFlowing && higherBlock.isFlowing
+            (inBlock.isFlowing && higherBlock.isFlowing)
```
Comment (0ffz) [inline src/main/kotlin/org/cultofclang/bonehurtingjuice/MoveListener.kt]: No need to add brackets

**Review by 0ffz — APPROVED** [review]
Most of this is fine, just the syntactical things, you think too low of yourself boy...

relevant xkcd:


Will merge after updating gradle stuff

File: src/main/kotlin/org/cultofclang/bonehurtingjuice/Extensions.kt (hunk 1 above)
Comment (Boy0000) [reply src/main/kotlin/org/cultofclang/bonehurtingjuice/Extensions.kt]: `internal val Block.isBubbleColumn: Boolean = blockData is BubbleColumn` didnt work but i did shorten it :)

**Pull request merged**